### PR TITLE
ARM MAP and minor changes

### PR DESCRIPTION
**Pull request by mand35**

### CODE REVIEW

**Review by pletzer — CHANGES_REQUESTED** [review]
Hi @mand35. Thanks for adding the ARM info.

1. 01-profiling.md line 19: "This can be a function, loop, or source code line based." (add article "a").

2. lines 23-24. I prefer the original version because by saying "The profiler can add significant overhead" this could be interpreted as "the numbers returned by the profiler are wrong". 

3. Line 38. Remove "Basically" and replace "spend" with "spent"

4. Line 40. Is it really true that the profiler adds significant overhead? If I'm not mistaken some profilers take into account their own overhead when returning timings. I'm wondering whether we want to mention this part.... 

5. Line 112. optimized -> optimised

6. There is another PR regarding the mpi section, which will be merged  sometime and may interfere with this PR. I suggest you create another branch and open another PR for the mpi section. Also we typically have a sentence at the top "Start by changing directory to blah blah". I understand it makes sense to change the directory after introducing the concepts but then this would be done in all lessons (a separate PR).

**Comment by mand35** [comment]
done

**Review by pletzer — COMMENTED** [review]
* Somehow we need to differentiate cProfile from MAP. 
  - When should cProfile be used and when should MAP be used? (cProfile can run on any serial Python code. It can be used to identify the functions that take most of the execution time. MAP is a commercial product installed on Mahuika, which can provide profiling information for any program, whether it is written Python or C/C++, and whether it is runs serially or in parallel. MAP supports MPI, OpenMP and Cuda. A graphical interface allows the user to focus on specific source lines, .... (list all the features here). Because MAP is a graphical tool, it can on the other hand slow over the network.

* I'm also thinking that how to use MAP is a big topic that deserves a separate lesson. We'll have to describe each step carefully with many screen shots.

* Keep in mind that we use English spelling - parrallelised instead of parallelized.

**Comment by mand35** [comment]
I added more details to the MAP description. Yes, I will add NeSI documentation for MAP. But I think for this context a short description on how to run map is sufficient. The GUI is user-friendly and fairly straight forward to use. It could be more of interest to interpret the presented data.

**Comment by pletzer** [comment]
Here are my recommendations:

1. Create a separate lesson 01b-profiling-with-map
2. Have Pros and Cons of MAP
3. Describe how to use MAP, including how to submit a parallel job. Include screen shots
4. Have an exercise section where student profiles the openmp version of scatter.

File: _lessons/python-scatter/01-profiling.md [hunk 1]
```diff
@@ -211,6 +212,17 @@ tell *memory_profiler* which functions you wish to profile.
...
 at the page linked above.
 
+## ARM MAP
+
+On NeSI systems another profiler is provide, the [ARM MAP](https://www.arm.com/products/development-tools/server-and-hpc/forge/map) (previous Allinea MAP) profiler, which is part of the `forge` module (as well as the parallel debugger DDT).
+
+In contrast to cProfile, which provides function based profiling information of Python programs, MAP is a commercial product, which can profile  parallel, multi-threaded and single-threaded C/C++, Fortran and F90, as well as Pyhton codes. MAP can be launched with a GUI, which allows the user to navigate thru the code and enables to focus on specific source lines, and also without a GUI for faster execution (see [documentation](https://developer.arm.com/docs/101136/latest/map)).
+
+The GUI can be started after loading `module load forge` and launching `map`. In the profile menu we need to specify the executable (in this case "python"), the arguments (here "scatter.py") and a working directory and if necessary the parallelisation parameters. After running the executable, MAP shows the summary with different metrics.
+[![example-map-scatter](images/MAP_scatter.png)](images/MAP_scatter.png)
```
Comment (chrisdjscott) [inline _lessons/python-scatter/01-profiling.md]: @mand35 the case of the MAP_scatter image is not correct in the link - you either need to rename the file from MAP_scatter.PNG -> MAP_scatter.png, or edit the link to make it MAP_scatter.PNG

**Comment by chrisdjscott** [comment]
Thanks @mand35

**Pull request merged**